test/instrumentation/decode_metric.go

```diff
@@ -160,14 +161,30 @@ func (c *metricDecoder) decodeOpts(expr ast.Expr) (metric, error) {
 
 		switch key {
 		case "Namespace", "Subsystem", "Name", "Help", "DeprecatedVersion":
-			k, ok := kv.Value.(*ast.BasicLit)
-			if !ok {
-				return m, newDecodeErrorf(expr, errNonStringAttribute)
-			}
-			if k.Kind != token.STRING {
+			var value string
+			var err error
+			switch v := kv.Value.(type) {
+			case *ast.BasicLit:
+				value, err = stringValue(v)
+				if err != nil {
+					return m, newDecodeErrorf(expr, errNonStringAttribute)
```

Comment:
Is there a specific reason you want to construct the error here from the expr as opposed to use the error returned from stringValue above?

Reply:
?

Reply:
After rethinking I changed it to return error from `stringValue`